### PR TITLE
fix: Add relevant fields to the Explore Catalog CSV Download

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

- follow-on to #375
- be more defensive with accessing algolia results (missing keys)
- ensure `key` and `content_type` are returned by algolia

## References

- [ENT-5295](https://openedx.atlassian.net/browse/ENT-5295)

### CODE REVIEW

File: enterprise_catalog/apps/api/v1/views.py [hunk 1]
```diff
@@ -384,17 +386,18 @@ def retrieve_indexed_data(self, facets, algoliaQuery):
...
+            if len(course_keys_chunk) > 0:
+                query_params = {'keys': ','.join(course_keys_chunk)}
+                courses = discovery_client.get_courses(query_params=query_params)
+                for course in courses:
+                    course_by_key[course.get('key')] = course
 
             # combine discovery metadata with the algolia results
             # append the hit to the results
-            for hit in page['hits']:
+            for hit in page.get('hits', []):
```
Comment (binodpant) [inline enterprise_catalog/apps/api/v1/views.py:400]: is there a test change that is needed to cover this? other than this, this is looking fine to me

File: enterprise_catalog/apps/api/v1/views.py [hunk 2]
```diff
@@ -210,6 +210,8 @@ class CatalogCsvDataView(GenericAPIView):
     ]
     algolia_attributes_to_retrieve = [
         'title',
+        'key',
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/views.py:213]: these attributes were not consistently being returned, which i think is the reason why i was seeing such strange differences between local, stage, and prod

File: enterprise_catalog/apps/api/v1/views.py (hunk 1 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/v1/views.py:400]: when do we expect no hits? When will page not have at least an empty hits attribute?

File: enterprise_catalog/apps/api/v1/views.py [hunk 3]
```diff
@@ -210,6 +210,8 @@ class CatalogCsvDataView(GenericAPIView):
...
+        'content_type',
```
Comment (alex-sheehan-edx) [inline enterprise_catalog/apps/api/v1/views.py:214]: oof, I probably should have had content_type here from the beginning 😬

**Review by alex-sheehan-edx — APPROVED** [review]
looks good to me, just a couple context questions/comments

File: enterprise_catalog/apps/api/v1/views.py (hunk 1 above)
Comment (johnnagro) [reply enterprise_catalog/apps/api/v1/views.py:400]: no, just being extra defensive in what we get back from algolia

**Pull request merged**